src/components/shared/basic/Input/ValidatingInput.tsx

```diff
@@ -51,11 +51,12 @@ const ValidatingInput = ({
 }: ValidatingInputProps) => {
   const [color, setColor] = useState<Colors>(Colors.secondary)
   const [currentValue, setCurrentValue] = useState<string>(value)
-
+  const [valid, setValid] = useState<boolean>(false)
   const immediateValidate = useCallback(
     (expr: string) => {
-      const isValid = validate ? validate(expr) : true
+      const isValid = validate?.(expr) ? true : false
```

Comment:
I think it is better to remove the optional marker for validate in line 32 because ValidatingInput doesn't make much sense without it. Then simply write:
`const isValid = validate(expr)`